staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1395,20 +1395,22 @@ func (e *Store) finalizeDelete(ctx context.Context, obj runtime.Object, runHooks
 // automatically.
 func (e *Store) Watch(ctx context.Context, options *metainternalversion.ListOptions) (watch.Interface, error) {
 	label := labels.Everything()
-	if options != nil && options.LabelSelector != nil {
+	if options == nil {
+		// By default we should serve the request from etcd.
+		options = &metainternalversion.ListOptions{ResourceVersion: ""}
+	}
+	if options.LabelSelector != nil {
 		label = options.LabelSelector
 	}
 	field := fields.Everything()
-	if options != nil && options.FieldSelector != nil {
+	if options.FieldSelector != nil {
 		field = options.FieldSelector
 	}
 	predicate := e.PredicateFunc(label, field)
 
-	resourceVersion := ""
-	if options != nil {
-		resourceVersion = options.ResourceVersion
-		predicate.AllowWatchBookmarks = options.AllowWatchBookmarks
-	}
+	resourceVersion := options.ResourceVersion
+	predicate.AllowWatchBookmarks = options.AllowWatchBookmarks
+
```

Comment:
It would be great if you could add a unit test that would fail with the previous code but passes with the updated code.

Reply:
+1